ElvUI/Modules/Maps/MapMarkers.lua

```diff
@@ -0,0 +1,189 @@
+	MM:SendCommMessage(prefix, text, distribution or "RAID");
+end
+local pname = UnitName("player");
+
+function MM:ReseiveMark(text, distribution, target)
+
+	local success,mapid, x, y, who = MM:Deserialize(text);
+	mapid = tonumber(mapid)
+	if success and mapid and x and y and who ~= pname and not IgnoreList[who] then
+		MM:PrintMarkInfo(mapid, x, y, who);
+	end
+end
+function MM:PrintMarkInfo(mapid, x, y, who)
+	DEFAULT_CHAT_FRAME:AddMessage(string.format(SYNC_INFO,who,mapid,x,y), 0.41, 0.8, 0.94);
+end
+function MM:HideAll()
+	for i = 1, #ElvUI_ShowedMarkers do
+		ElvUI_ShowedMarkers[i]:Hide();
+	end
+	table.wipe(ElvUI_ShowedMarkers);
+end
+
+function MM:ShowMark(i,mapid)
+	local marker = ElvUI_AllMarkers[mapid][i];
+
+	if marker.showed then
+		marker:Show()
+		table.insert(ElvUI_ShowedMarkers, marker);
+	elseif not marker.showed then
+		marker:Hide()
+	end
+
+end
+
+function MM:CreateMark(mapid,IsSendedMark,x,y)
+
+	mapid = mapid or GetCurrentMapAreaID();
+	mapid = tonumber(mapid)
+	if not WorldMapFrame:IsShown() and IsSendedMark then
+		ToggleFrame(WorldMapFrame)
+		SetMapByID(mapid-1)
+	end
+	local curX, curY = GetCursorPosition();
+	local scale = WorldMapDetailFrame:GetEffectiveScale();
+	local width, height = WorldMapDetailFrame:GetSize();
+	local centerX, centerY = WorldMapDetailFrame:GetCenter();
+	local adjustedX = (curX / scale - (centerX - (width * 0.5))) / width;
+	local adjustedY = (centerY + (height * 0.5) - curY / scale) / height;
+	x = IsSendedMark and x or adjustedX * 100;
+	y = IsSendedMark and y or adjustedY * 100;
+	ElvUI_AllMarkers[mapid] = ElvUI_AllMarkers[mapid] or {};
+
+	local lastIndex = #ElvUI_AllMarkers[mapid];
+	ElvUI_AllMarkers[mapid][lastIndex+1] = CreateFrame("Frame", "Marker"..mapid..lastIndex+1, WorldMapDetailFrame);
+	local marker = ElvUI_AllMarkers[mapid][lastIndex+1];
+	marker.name = "Marker"..mapid..lastIndex;
+	marker.index = lastIndex+1;
+	marker.x = x;
+	marker.y = y;
+	marker.Texture = marker:CreateTexture();
+	marker.Texture:SetTexture(texturePath);
+	marker.Texture:SetAllPoints();
+	marker.Texture:SetTexCoord(unpack(coords))
+	marker:SetPoint("CENTER", WorldMapDetailFrame, "TOPLEFT", (x / 100) * WorldMapDetailFrame:GetWidth(), (-y / 100) * WorldMapDetailFrame:GetHeight());
+	marker:SetWidth(31);
+	marker:SetHeight(31);
+	marker:SetFrameStrata("DIALOG");
+	marker:EnableMouse(true);
+	marker:HookScript("OnEnter", function(self)
+		GameTooltip:SetOwner(self, "ANCHOR_LEFT")
+		GameTooltip:SetText("SHIFT + ПКМ убрать метку", 1, 1, 1)
+		GameTooltip:Show()
+	end)
+	marker:SetScript("OnLeave", function()
+		GameTooltip:Hide()
+	end)
+	marker:SetScript("OnMouseDown",function(self,click)
+		if IsShiftKeyDown() and click == "MiddleButton" then
+			self:Hide();
+			self.showed = false;
+			MM:RefreshAll();
+		end
+	end)
+	marker.showed = true;
+	table.insert(ElvUI_ShowedMarkers, marker);
+	marker:Show();
+	if IsSendedMark then
+		print("Marker whit MapID "..mapid .." x ".. x .." y ".. y .." was added on map");
```

Comment:
Is it important the player gets ID / x / y? Maybe could do
```lua
print(format("Map marker added on %s at (%.2f, %.2f)", GetMapName(mapID), x, y))
```

Reply:
Maybe not
Il do 
print(format("Map marker added in %s at (%.1f, %.1f)", GetMapName(mapID), x, y))